src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java

```diff
@@ -14,14 +15,18 @@
 
 import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.COMPANY_NUMBER;
 import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.ERIC_REQUEST_ID_KEY;
-import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.MAX_COMPANY_NUMBER_LENGTH;
-import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.MAX_ID_LENGTH;
 
 @Component
 public class CompanyNumberValidationInterceptor implements HandlerInterceptor {
 
-    private static final Pattern COMPANY_NUMBER_PATTERN = Pattern.compile(
-            "^([a-z]|[a-z][a-z])?\\d{6,8}$", Pattern.CASE_INSENSITIVE);
+    @Value("${MAX_ID_LENGTH}")
+    private String maxIdLengthString;
+
+    @Value("${MAX_COMPANY_NUMBER_LENGTH}")
+    private String maxComapnyNumberLengthString;
+
+    @Value("${COMPANY_NUMBER_PATTERN}")
+    private String confirmationNumberPattern;
```

Comment:
This var should be named companyNumberPattern, the confirmation number implies it's the confirmation statement id that we are checking